### ISSUE TITLE
Running Mask R-CNN in the development build

### ISSUE THREAD

**Issue by luxedo** [jsbroks/coco-annotator#199]
I'm trying to run the `Mask R-CNN` model in the development build. I can run the model in the production build. 

I downloaded the model and set the environment variable in `docker-compose.dev.yml`. 

When I start the development build `$ docker-compose -f docker-compose.dev.yml up --build`, the page won`t load and I keep getting the following message:

```
annotator_webserver | Building network based on ResNet 101 and PSP module expecting inputs of shape (512, 512) predicting 1 classes
annotator_webserver | PSP module will interpolate to a final feature map size of (64, 64)
annotator_webserver | [2019-04-25 18:59:30 +0000] [564] [INFO] Booting worker with pid: 564
annotator_webserver | Using TensorFlow backend.
annotator_webserver | WARNING:tensorflow:From /usr/local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
annotator_webserver | Instructions for updating:
annotator_webserver | Colocations handled automatically by placer.
annotator_webserver | WARNING:tensorflow:From /usr/local/lib/python3.6/site-packages/mask_rcnn-2.1-py3.6.egg/mrcnn/model.py:772: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
annotator_webserver | Instructions for updating:
annotator_webserver | Use tf.cast instead.
annotator_webserver | 2019-04-25 18:59:37.515529: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
annotator_webserver | 2019-04-25 18:59:37.538545: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2304000000 Hz
annotator_webserver | 2019-04-25 18:59:37.539099: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x560fa0c6b090 executing computations on platform Host. Devices:
annotator_webserver | 2019-04-25 18:59:37.539116: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
annotator_webserver | [2019-04-25 19:00:00 +0000] [6] [CRITICAL] WORKER TIMEOUT (pid:564)
annotator_webserver | [2019-04-25 19:00:00 +0000] [564] [INFO] Worker exiting (pid: 564)
annotator_webserver | Building network based on ResNet 101 and PSP module expecting inputs of shape (512, 512) predicting 1 classes
annotator_webserver | PSP module will interpolate to a final feature map size of (64, 64)
```

I tried to run in different tags (v0.10.0, v0.10.2, v0.10.4) and they also gives the same error. I tried to increase the timeout to 5 minutes and got no erros , but still can`t load the page.

**Comment by jsbroks** [comment]
Very strange. Both production and development boot up the model the same way. How excatly are you setting the timeout in development?

**Comment by luxedo** [comment]
You're right, I think I'm not managing the docker images correctly. I just pruned all the images and will test again.

I set the timeout in `backend/webserver/gunicorn_config.py`.

**Comment by luxedo** [comment]
Yeah, after a clean build I can`t even run the production build.
My setup is the following: 
- `master` without modifications ([`8f336177c987bfd9d17620f95fc055c5a3b0e289`](https://github.com/jsbroks/coco-annotator/commit/8f336177c987bfd9d17620f95fc055c5a3b0e289))
- no models inside `models/`

In the production build I get this error:
```
annotator_webclient | [2019-04-26 14:46:33 +0000] [11] [ERROR] Exception in worker process
annotator_webclient | Traceback (most recent call last):
annotator_webclient |   File "/usr/local/lib/python3.6/site-packages/gunicorn/arbiter.py", line 583, in spawn_worker
annotator_webclient |     worker.init_process()
annotator_webclient |   File "/usr/local/lib/python3.6/site-packages/gunicorn/workers/geventlet.py", line 102, in init_process
annotator_webclient |     super(EventletWorker, self).init_process()
annotator_webclient |   File "/usr/local/lib/python3.6/site-packages/gunicorn/workers/base.py", line 129, in init_process
annotator_webclient |     self.load_wsgi()
annotator_webclient |   File "/usr/local/lib/python3.6/site-packages/gunicorn/workers/base.py", line 138, in load_wsgi
annotator_webclient |     self.wsgi = self.app.wsgi()
annotator_webclient |   File "/usr/local/lib/python3.6/site-packages/gunicorn/app/base.py", line 67, in wsgi
annotator_webclient |     self.callable = self.load()
annotator_webclient |   File "/usr/local/lib/python3.6/site-packages/gunicorn/app/wsgiapp.py", line 52, in load
annotator_webclient |     return self.load_wsgiapp()
annotator_webclient |   File "/usr/local/lib/python3.6/site-packages/gunicorn/app/wsgiapp.py", line 41, in load_wsgiapp
annotator_webclient |     return util.import_app(self.app_uri)
annotator_webclient |   File "/usr/local/lib/python3.6/site-packages/gunicorn/util.py", line 350, in import_app
annotator_webclient |     __import__(module)
annotator_webclient |   File "/workspace/webserver/__init__.py", line 62, in <module>
annotator_webclient |     app = create_app()
annotator_webclient |   File "/workspace/webserver/__init__.py", line 57, in create_app
annotator_webclient |     ImageModel.objects.update(annotating=[])
annotator_webclient |   File "/usr/local/lib/python3.6/site-packages/mongoengine/queryset/manager.py", line 37, in __get__
annotator_webclient |     queryset = queryset_class(owner, owner._get_collection())
annotator_webclient |   File "/usr/local/lib/python3.6/site-packages/mongoengine/document.py", line 208, in _get_collection
annotator_webclient |     db.client.is_primary:
annotator_webclient |   File "/usr/local/lib/python3.6/site-packages/pymongo/mongo_client.py", line 857, in is_primary
annotator_webclient |     return self._server_property('is_writable')
annotator_webclient |   File "/usr/local/lib/python3.6/site-packages/pymongo/mongo_client.py", line 689, in _server_property
annotator_webclient |     writable_server_selector)
annotator_webclient |   File "/usr/local/lib/python3.6/site-packages/pymongo/topology.py", line 224, in select_server
annotator_webclient |     address))
annotator_webclient |   File "/usr/local/lib/python3.6/site-packages/pymongo/topology.py", line 183, in select_servers
annotator_webclient |     selector, server_timeout, address)
annotator_webclient |   File "/usr/local/lib/python3.6/site-packages/pymongo/topology.py", line 199, in _select_servers_loop
annotator_webclient |     self._error_message(selector))
annotator_webclient | pymongo.errors.ServerSelectionTimeoutError: No servers found yet
annotator_webclient | [2019-04-26 14:46:33 +0000] [11] [INFO] Worker exiting (pid: 11)
annotator_webclient | [2019-04-26 14:46:34 +0000] [7] [INFO] Shutting down: Master
annotator_webclient | [2019-04-26 14:46:34 +0000] [7] [INFO] Reason: Worker failed to boot.
annotator_webclient exited with code 3
```
And this error for the development build:
```
annotator_webserver | [2019-04-26 14:52:50 +0000] [10] [CRITICAL] WORKER TIMEOUT (pid:14)
annotator_webserver | [2019-04-26 14:52:50 +0000] [14] [INFO] Worker exiting (pid: 14)
annotator_webserver | [2019-04-26 14:52:51 +0000] [30] [INFO] Booting worker with pid: 30
annotator_webserver | [2019-04-26 14:52:51 +0000] [30] [WARNING] MaskRCNN model is disabled.
annotator_webserver | [2019-04-26 14:52:51 +0000] [30] [WARNING] DEXTR model is disabled.
annotator_webserver | [2019-04-26 14:53:21 +0000] [10] [CRITICAL] WORKER TIMEOUT (pid:30)
annotator_webserver | [2019-04-26 14:53:21 +0000] [30] [INFO] Worker exiting (pid: 30)
annotator_webserver | [2019-04-26 14:53:22 +0000] [46] [INFO] Booting worker with pid: 46
annotator_webserver | [2019-04-26 14:53:22 +0000] [46] [WARNING] MaskRCNN model is disabled.
annotator_webserver | [2019-04-26 14:53:22 +0000] [46] [WARNING] DEXTR model is disabled.
annotator_webserver | [2019-04-26 14:53:52 +0000] [10] [CRITICAL] WORKER TIMEOUT (pid:46)
annotator_webserver | [2019-04-26 14:53:52 +0000] [46] [INFO] Worker exiting (pid: 46)
annotator_webserver | [2019-04-26 14:53:53 +0000] [62] [INFO] Booting worker with pid: 62
annotator_webserver | [2019-04-26 14:53:54 +0000] [62] [WARNING] MaskRCNN model is disabled.
annotator_webserver | [2019-04-26 14:53:54 +0000] [62] [WARNING] DEXTR model is disabled.
annotator_webserver | [2019-04-26 14:54:23 +0000] [10] [CRITICAL] WORKER TIMEOUT (pid:62)
annotator_webserver | [2019-04-26 14:54:23 +0000] [62] [INFO] Worker exiting (pid: 62)
annotator_webserver | [2019-04-26 14:54:24 +0000] [78] [INFO] Booting worker with pid: 78
annotator_webserver | [2019-04-26 14:54:25 +0000] [78] [WARNING] MaskRCNN model is disabled.
annotator_webserver | [2019-04-26 14:54:25 +0000] [78] [WARNING] DEXTR model is disabled.
annotator_webserver | [2019-04-26 14:54:54 +0000] [10] [CRITICAL] WORKER TIMEOUT (pid:78)
annotator_webserver | [2019-04-26 14:54:54 +0000] [78] [INFO] Worker exiting (pid: 78)
annotator_webserver | [2019-04-26 14:54:55 +0000] [94] [INFO] Booting worker with pid: 94
annotator_webserver | [2019-04-26 14:54:56 +0000] [94] [WARNING] MaskRCNN model is disabled.
annotator_webserver | [2019-04-26 14:54:56 +0000] [94] [WARNING] DEXTR model is disabled.
annotator_webserver | [2019-04-26 14:55:25 +0000] [10] [CRITICAL] WORKER TIMEOUT (pid:94)
annotator_webserver | [2019-04-26 14:55:25 +0000] [94] [INFO] Worker exiting (pid: 94)
```

**Comment by jsbroks** [comment]
Looks like the webserver is unable to connect to the database.
What version did you upgrade from? How did you upgrade?

**Comment by luxedo** [comment]
I wanted to test a custom Mask R-CNN model with 3 classes. I rebuilt the images after cleaning all with 
```
$ docker system prune -a
```

I tried again, pruning everything and building. Both `master` and `v0.10.4` worked. I'm very puzzled. I'm no expert in docker, maybe some cached images were messing my setup.

**Comment by luxedo** [comment]
I managed to make my model work by setting `timeout = 60` in `gunicorn_config.py`. 

I think that the model was taking too long to load, then the webserver was killing the model process and restarting it. I still don't understand why it didn't worked with a timeout of 5 minutes, maybe that's something else.

I made a PR that didn't included this change but fixed a couple of things I had to do to run my model: https://github.com/jsbroks/coco-annotator/pull/201
